client/src/sagas/auth.js

```diff
@@ -1,20 +1,58 @@
-import { delay, put, takeLatest } from 'redux-saga/effects';
-import { LOGIN, LOGIN_ASYNC, REGISTER, REGISTER_ASYNC } from './types';
+import { push } from 'connected-react-router';
+import { put, takeLatest } from 'redux-saga/effects';
+import axios from 'axios';
 
-export function* loginAsync({ payload }) {
-  yield delay(1000);
-  yield put({ type: LOGIN, payload });
-}
+import { auth } from './types';
+import { alertActions } from './actions';
 
-export function* registerAsync({ payload }) {
-  yield delay(1000);
-  yield put({ type: REGISTER, payload });
+// logout
+export function* watchLogout() {
+  yield takeLatest(auth.LOGOUT, logout);
+}
+export function* logout() {
+  yield put({ type: auth.reducer.LOGOUT });
 }
 
+// login
 export function* watchLoginAsync() {
-  yield takeLatest(LOGIN_ASYNC, loginAsync);
+  yield takeLatest(auth.LOGIN, loginAsync);
+}
+export function* loginAsync({ payload }) {
+  try {
+    const res = yield axios.get(`/player/${payload}`);
+    yield put({
+      type: auth.reducer.LOGIN,
+      payload: { id: res.data.id, name: res.data.name },
+    });
+    yield put(alertActions.addAlert('Login successful!', 'success'));
+    yield put(push('/home'));
+  } catch (err) {
+    yield put({
+      type: auth.reducer.LOGIN_FAILED,
+      payload: err.response.data,
+    });
+    yield put(alertActions.addAlert(err.response.data, 'error'));
+  }
 }
 
+// register
 export function* watchRegisterAsync() {
-  yield takeLatest(REGISTER_ASYNC, registerAsync);
+  yield takeLatest(auth.REGISTER, registerAsync);
+}
+export function* registerAsync({ payload: { id, name } }) {
+  try {
+    const res = yield axios.post('/create/player', { id, name });
+    yield put({
+      type: auth.reducer.REGISTER,
+      payload: { id: res.data.id, name: res.data.name },
+    });
+    yield put(alertActions.addAlert('Registration successful!', 'success'));
+    yield put(push('/home'));
```

Comment:
So on successful registration ... does this push us to the "home" page? Shouldn't that be the "user's home page"? Is that what's happening?

Reply:
Yes, the home page is a logged in user's home page.